examples/scale/scenarios/scenarios.go

```diff
@@ -48,6 +88,13 @@ type Scenario struct {
 	Evaluator evaluatorFunction
 }
 
+type LogicalScenario interface {
```

Comment:
Naming suggestion above - but can you please add more comments over these interfaces to add clarity?  Also moving the interface declaration above its first usage in the file will help.